### ISSUE TITLE
Add hook to call on missing properties

### ISSUE THREAD

**Issue by JuergenSimon** [wycats/handlebars.js#1435]
Before filing issues, please check the following points first:

- [X] Have a look at https://github.com/wycats/handlebars.js/blob/master/CONTRIBUTING.md
- [X] Read the FAQ at https://github.com/wycats/handlebars.js/blob/master/FAQ.md
- [X] Use the jsfiddle-template at https://jsfiddle.net/9D88g/47/ to reproduce problems or bugs

Check out the jsfiddle at https://jsfiddle.net/fwrww8tj. What is demonstrated there is this problem: when registering a helper for catching missing keys in the data using:

`Handlebars.registerHelper("helperMissing", function(context, options) { ... })`

the helper function is called correctly when keys are missing on the top-level, but fails when the key is down the path in the data somewhere. Consider the following data set: 

```
{
    foo: "foo",
    nested_property: {
        foo: "foo"
    }
}
```

When the template calls for `{{bar}}`, the function is correctly called. However, when the template calls for {{nested_property.bar}}, the helper is NOT called.

**Comment by nknapp** [comment]
nested_property.bar is not a helper call, but a property access. When accessing top-level properties, there is no syntactical distinction between helpers and property access, but helpers cannot contain a ".", as far as I know. See if you can use the " with"

**Comment by jimmyn** [comment]
I'm having the same issue, I need to warn a user that property does not exist on some object but so far I haven't found a way to do that

**Comment by nknapp** [comment]
Changing this probably reaches down into compiler. I am not sure what exactly needs to be changed.

@wycats What do you think of this? A hook to be called when nested properties don't exist?

**Comment by jimmyn** [comment]
@nknapp @wycats any news on this? Maybe you can suggest some workaround?

**Comment by nknapp** [comment]
no, it makes sense, but I haven't got the time to work on it and it seems like @wycats is also busy.

Besides, we want a spec (#1277) before adding new features. Before I start considering it, I want @wycats approval, because he explicitly asked me only to fix bugs and not to add new features until we have a spec.

**Comment by kohlerdominik** [comment]
Three years and counting :)
I think it would be a good addiction to provide a `propertyMissing` hook.

I want to share our usecase for this:
We provide a TemplateEditor, shown as HTML/Mustache-Input on the left side and a rendered preview on the right side.
It's easier to use if the user has not to provide a context while writing the template, so we'd like to render the property names instead of the property values in the template.

What we would like to see is something like:
```js
Handlebars.registerPropertyMissingHandler(function(name) { 
  return "{{" + name + "}}"
})
```

**Comment by stevemk14ebr** [comment]
strict mode doesn't properly throw when dotted lookups occur. This breaks my workflow when rendering templates, expecting ones with missing fields/properties to fail.

```
HandleBars.compile(t, { strict: true })
```

```
{#each rollup}} | `{{{this.[original filename]}}}` | `{{{this.size}}}` |
{{/each}}
```

**Comment by nliu-ms** [comment]
Any update on this? It's a valuable feature for us as well.

**Comment by Mupli** [comment]
still no update on this ?? lol

**Comment by misabel** [comment]
+ 1 to this being a valuable feature addition

**Comment by geoffreygarrett** [comment]
RIP

**Comment by Ben-Lawrencee** [comment]
+1 indeed. Please implement this valuable feature.
